modules/programs/notmuch.nix

```diff
@@ -164,7 +170,7 @@ in {
       }
     ];
 
-    home.packages = [ pkgs.notmuch ];
+    home.packages = [ cfg.package ];
```

Comment:
The line

    export PATH="${pkgs.notmuch}/bin''${PATH:+:}$PATH"

below also needs to be updated. Same with

    ''"PATH=${pkgs.notmuch}/bin"''

in `muchsync.nix` and probably also

    "'${pkgs.notmuch}/bin/notmuch address --format=json --output=recipients  date:6M..'";

in `alot-accounts.nix` (although that one is probably a bit trickier to do since it's in a submodule).

In general I wonder if it is not more reliable to use an overlay if you want to use a custom notmuch package? The notmuch package may show up in many places and with an overlay you can make certain that all the references to `pkgs.notmuch` point to your desired package.

Reply:
> (although that one is probably a bit trickier to do since it's in a submodule).

Wouldn't using `${config.programs.notmuch.package}/bin/notmuch` works there?

> In general I wonder if it is not more reliable to use an overlay if you want to use a custom notmuch package?

That's what I'm currently doing for my own use, but I thought putting it as an option could be of interest for other people.